src/test/java/fi/metatavu/metaform/server/test/functional/tests/VersionTestsIT.kt

```diff
@@ -300,4 +300,88 @@ class VersionTestsIT : AbstractTest() {
             )
         }
     }
+
+    @Test
+    @Throws(Exception::class)
+    fun updateVersion() {
+        TestBuilder().use { testBuilder ->
+            val metaform = testBuilder.systemAdmin.metaforms.createFromJsonFile("simple")
+            val version = MetaformVersion(
+                type = MetaformVersionType.DRAFT,
+                data = readJsonMap(objectMapper.writeValueAsString(metaform))
+            )
+            val createdVersion = testBuilder.systemAdmin.metaformVersions.create(
+                metaformId = metaform.id!!,
+                payload = version
+            )
+            val versionModified = createdVersion.copy(
+                type = MetaformVersionType.ARCHIVED
+            )
+
+            testBuilder.systemAdmin.metaformVersions.updateMetaformVersion(
+                metaformId = metaform.id,
+                versionId = createdVersion.id!!,
+                body = versionModified
+            )
+            val updatedVersion = testBuilder.systemAdmin.metaformVersions.findVersion(
+                metaformId = metaform.id,
+                versionId = createdVersion.id
+            )
+
+            Assertions.assertEquals(createdVersion.createdAt, updatedVersion.createdAt)
+            Assertions.assertEquals(createdVersion.creatorId, updatedVersion.creatorId)
+            Assertions.assertEquals(createdVersion.lastModifierId, updatedVersion.lastModifierId)
+            Assertions.assertEquals(createdVersion.data, updatedVersion.data)
+            Assertions.assertNotEquals(createdVersion.type, updatedVersion.type)
+            Assertions.assertNotEquals(createdVersion.modifiedAt, updatedVersion.modifiedAt)
+        }
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun updateVersionUnAuthorized() {
```

Comment:
Please use the testbuilder function permissionTestByScopes to test the permission in different scopes